packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx

```diff
@@ -68,99 +74,168 @@ export default function CreatePrefabPanel({ entity }: { entity: Entity }) {
+      ? defaultPrefabFolder.value + '/' + prefabName.value + '.lookdev' + '.gltf'
+      : defaultPrefabFolder.value + '/' + prefabName.value + '.gltf'
     const srcProject = editorState.projectName!
     const fileURL = pathJoin(config.client.fileServer, 'projects', srcProject, fileName)
+
     try {
-      const parentEntity = getComponent(entity, EntityTreeComponent).parentEntity
       const resourcesold = await API.instance.service(staticResourcePath).find({
         query: { key: 'projects/' + srcProject + '/' + fileName }
       })
       if (resourcesold.data.length !== 0 && !isOverwriteConfirmed.value) {
         console.log('this name already exist, click confirm to overwrite the prefab')
         await isOverwriteModalVisible.set(true)
       } else {
-        const prefabEntity = createEntity()
-        const obj = new Scene()
-        addObjectToGroup(prefabEntity, obj)
-        proxifyParentChildRelationships(obj)
-        setComponent(prefabEntity, EntityTreeComponent, { parentEntity })
-        setComponent(prefabEntity, NameComponent, prefabName.value)
-        const entityTransform = getComponent(entity, TransformComponent)
-        const position = entityTransform.position.clone()
-        const rotation = entityTransform.rotation.clone()
-        const scale = entityTransform.scale.clone()
-        setComponent(prefabEntity, TransformComponent, {
-          position,
-          rotation,
-          scale
-        })
-        setComponent(entity, TransformComponent, {
-          position: new Vector3(0, 0, 0),
-          rotation: new Quaternion().identity(),
-          scale: new Vector3(1, 1, 1)
-        })
-        setComponent(entity, EntityTreeComponent, { parentEntity: prefabEntity })
-        getMutableState(SelectionState).selectedEntities.set([])
-        getComponent(entity, TransformComponent).matrix.identity()
-        await exportRelativeGLTF(prefabEntity, srcProject, fileName)
-
-        const resources = await API.instance.service(staticResourcePath).find({
-          query: { key: 'projects/' + srcProject + '/' + fileName }
-        })
-        if (resources.data.length === 0) {
-          throw new Error('User not found')
+        if (isExportLookDev) {
+          const lookdevEntity = [] as Entity[]
+          const lookDevComponent: Component[] = [
+            SkyboxComponent,
+            HemisphereLightComponent,
+            DirectionalLightComponent,
+            PostProcessingComponent
+          ]
+          const prefabEntity = createEntity()
+          const obj = new Scene()
+          addObjectToGroup(prefabEntity, obj)
+          proxifyParentChildRelationships(obj)
+          const rootEntity = getState(EditorState).rootEntity
+          iterateEntityNode(rootEntity, (entity) => {
+            lookDevComponent.forEach((component) => {
+              if (hasComponent(entity, component)) {
+                lookdevEntity.push(entity)
+              }
+            })
+          })
```

Comment:
this is nice. Add a check if lookdevEntity already has this component somewhere.